lib/matplotlib/axis.py

```diff
@@ -1023,6 +1023,97 @@ def set_default_intervals(self):
         # attribute, and the derived code below will check for that
         # and use it if it's available (else just use 0..1)
 
+    def _set_lim(self, v0, v1, alt0=None, alt1=None, *,
+                 emit=True, auto, names=("", "")):
+        """
+        Set view limits.
+
+        This method is a helper for the Axes ``set_xlim``, ``set_ylim``, and
+        ``set_zlim`` methods.  This docstring uses names corresponding to
+        ``set_xlim`` for simplicity.
+
+        *names* is the pair of the names of the first two parameters of the
+        Axes method (e.g., "left" and "right").  They are only used to generate
+        error messages; and can be empty if the limits are known to be valid.
+
+        Other parameters are directly forwarded from the Axes limits setter:
+        *v0*, *v1*, *alt0*, and *alt1* map to *left*, *right*, *xmin*, and
+        *xmax* respectively; *emit* and *auto* are used as is.
+        """
+        v0name, v1name = names  # The value names.
+        name, = [name for name, axis in self.axes._get_axis_map().items()
+                 if axis is self]  # The axis name.
+
+        if v1 is None and np.iterable(v0):
+            v0, v1 = v0
+        if alt0 is not None:
+            if v0 is not None:
+                raise TypeError(
+                    f"Cannot pass both {v0name!r} and '{name}lim'")
+            v0 = alt0
+        if alt1 is not None:
+            if v1 is not None:
+                raise TypeError(
+                    f"Cannot pass both {v1name!r} and '{name}lim'")
+            v1 = alt1
+
+        self.axes._process_unit_info([(name, (v0, v1))], convert=False)
+        v0 = self.axes._validate_converted_limits(v0, self.convert_units)
+        v1 = self.axes._validate_converted_limits(v1, self.convert_units)
+
+        if v0 is None or v1 is None:
+            # Axes init calls set_xlim(0, 1) before get_xlim() can be called,
+            # so only grab the limits if we really need them.
+            old0, old1 = self.get_view_interval()
+            if v0 is None:
+                v0 = old0
+            if v1 is None:
+                v1 = old1
+
+        if self.get_scale() == 'log' and (v0 <= 0 or v1 <= 0):
+            # Axes init calls set_xlim(0, 1) before get_xlim() can be called,
+            # so only grab the limits if we really need them.
+            old0, old1 = self.get_view_interval()
+            if v0 <= 0:
+                _api.warn_external(
+                    f"Attempt to set non-positive {v0name} {name}lim on a "
+                    f"log-scaled axis will be ignored.")
```

Comment:
Is {v0name} really needed here?

> Attempt to set non-positive xlim on a log-scaled axis

is almost as good as 

> Attempt to set non-positive left xlim on a log-scaled axis

I suppose the user can still identify where the problem is.